pgmpy/tests/test_models/test_NaiveBayes.py

```diff
@@ -0,0 +1,193 @@
+        self.assertRaises(ValueError, self.G.add_edges_from, [('a', 'e'), ('b', 'f')])
+        self.assertRaises(ValueError, self.G.add_edges_from, [('b', 'f')])
+
+    def test_add_edge_nonstring(self):
+        self.G.add_edge(1, 2)
+        six.assertCountEqual(self, self.G.nodes(), [1, 2])
+        self.assertListEqual(self.G.edges(), [(1, 2)])
+        self.assertEqual(self.G.parent_node, 1)
+        self.assertSetEqual(self.G.children_nodes, {2})
+
+        self.G.add_nodes_from([3, 4])
+        self.G.add_edge(1, 3)
+        self.G.add_edge(1, 4)
+        six.assertCountEqual(self, self.G.nodes(), [1, 2, 3, 4])
+        six.assertCountEqual(self, self.G.edges(), [(1, 2), (1, 3), (1, 4)])
+        self.assertEqual(self.G.parent_node, 1)
+        self.assertSetEqual(self.G.children_nodes, {2, 3, 4})
+
+        self.assertRaises(ValueError, self.G.add_edge, 2, 3)
+        self.assertRaises(ValueError, self.G.add_edge, 3, 6)
+        self.assertRaises(ValueError, self.G.add_edge, 5, 6)
+        self.assertRaises(ValueError, self.G.add_edges_from, [(1, 5), (2, 6)])
+        self.assertRaises(ValueError, self.G.add_edges_from, [(2, 6)])
+
+    def test_add_edge_selfloop(self):
+        self.assertRaises(ValueError, self.G.add_edge, 'a', 'a')
+        self.assertRaises(ValueError, self.G.add_edge, 1, 1)
+
+    def test_add_edges_from_self_loop(self):
+        self.assertRaises(ValueError, self.G.add_edges_from,
+                          [('a', 'a')])
+
+    def test_update_node_parents_bm_constructor(self):
+        self.g = NaiveBayes([('a', 'b'), ('a', 'c')])
+        self.assertListEqual(self.g.predecessors('a'), [])
+        self.assertListEqual(self.g.predecessors('b'), ['a'])
+        self.assertListEqual(self.g.predecessors('c'), ['a'])
+
+    def test_update_node_parents(self):
+        self.G.add_nodes_from(['a', 'b', 'c'])
+        self.G.add_edges_from([('a', 'b'), ('a', 'c')])
+        self.assertListEqual(self.G.predecessors('a'), [])
+        self.assertListEqual(self.G.predecessors('b'), ['a'])
+        self.assertListEqual(self.G.predecessors('c'), ['a'])
+
+    def tearDown(self):
+        del self.G
+
+class TestNaiveBayesMethods(unittest.TestCase):
+    def setUp(self):
+        self.G = NaiveBayes([('a', 'b'), ('a', 'c'),
+                             ('a', 'd'), ('a', 'e')])
+
+    def test_local_independencies(self):
+        self.assertListEqual(self.G.local_independencies('a'), [None])
+        self.assertListEqual(self.G.local_independencies('b'), 
+                            [Independencies(['b', ['e','c','d'], 'a'])])
+        self.assertListEqual(self.G.local_independencies('c'), 
+                            [Independencies(['c', ['e','b','d'], 'a'])])
+        self.assertListEqual(self.G.local_independencies('d'),
+                            [Independencies(['d', ['b','c','e'], 'a'])])
+        
+    def tearDown(self):
+        del self.G
+        
+class TestNaiveBayes_active_trail_nodes(unittest.TestCase):
```

Comment:
Can't you put these in `TestNaiveBayesMethods`  since it's `active_trail_nodes`?